desk/app/grouper.hoon

```diff
@@ -104,30 +126,48 @@
   ?-  -.sign
       %poke-ack   `this
       %watch-ack  `this
-      %kick       
+      %kick
     :_  this
     ~[(bite-subscribe bowl)]
     ::
       %fact
     =+  !<(=bite:reel q.cage.sign)
-    ~?  dev-mode  [bite (~(has in enabled-groups) token.bite)]
-    ?>  (~(has in enabled-groups) token.bite)
-    ?>  ?=([%bite-1 *] bite)
-    ~?  dev-mode  'inviting'
-    =/  =invite:groups  [[our.bowl token.bite] joiner.bite]
+    ?>  ?=([%bite-2 *] bite)
     :_  this
-    =/  our  (scot %p our.bowl)
-    =/  =path  /[our]/groups/(scot %da now.bowl)/groups/[our]/[token.bite]/noun
+    =;  caz=(list card)
+      =/  wir=^wire  /dm/(scot %p joiner.bite)
+      =/  =dock  [our.bowl %chat]
+      =/  =id:c  [our now]:bowl
+      =/  =memo:c
+        [~[[%inline ~[[%ship joiner.bite] ' has joined the network']]] id]
+      =/  =action:dm:c
+        :-  joiner.bite
+        [id %add memo [%notice ~] ~]
+      =/  =cage  chat-dm-action+!>(`action:dm:c`action)
+      (welp caz ~[[%pass wir %agent dock %poke cage]])
+    =;  result=(each (list card) (list tank))
+      ?:  ?=(%.y -.result)  p.result
+      ((slog p.result) ~)
+    %-  mule
+    |.
+    ?~  group=(~(get by fields.metadata.bite) 'group')
+      ~|("no group field" !!)
+    =/  =flag:groups  (flag:dejs:gj s+u.group)
+    ~?  dev-mode  [bite (~(has in enabled-groups) q.flag)]
+    ?>  (~(has in enabled-groups) q.flag)
+    ~?  dev-mode  'inviting'
+    =/  =invite:groups  [flag joiner.bite]
+    =/  =path  /(scot %p our.bowl)/groups/(scot %da now.bowl)/groups/(scot %p p.flag)/[q.flag]/noun
     =+  .^(=group:groups %gx path)
```

Comment:
Are you virtualizing here to save yourself if the scry crashes? Unfortunately, that won't save you. You'll want to do an explicit existence check.

Usually that'd be a `%gu` scry, but groups appears to implement a `%gx /exists/~some/group/noun`.

Reply:
Was virtualizing here for the early return if we don't have a group, but I guess I could just return empty list in both crash/assertions